server.ts

```diff
@@ -58,9 +58,9 @@ app.get("/", (req, res) => {
   } else res.redirect("/survey");
 });
 
-const getsurvey = async (urlstr: string, req: Request<{}>, res: Response<any>) =>  {
+const getsurvey = async (query: string | ParsedQs, req: Request<{}>, res: Response<any>) =>  {
   try {
-    const survey_url = new URL(urlstr);
+    const survey_url = new URL(query['url']);
     res.render("survey", {
       query: req.query,
```

Comment:
Shouldn't this be something like: 
```suggestion
      query: query,
```

Reply:
Whoops. I fixed it in the latest commit.